lte/gateway/python/integ_tests/s1aptests/test_continuous_random_attach.py

```diff
@@ -0,0 +1,188 @@
+            s1ap_types.ueDetachType_t.UE_SWITCHOFF_DETACH.value
+        )
+        self._s1ap_wrapper._s1_util.issue_cmd(
+            s1ap_types.tfwCmd.UE_DETACH_REQUEST, detach_req
+        )
+
+    def ue_thread_function(self, req, queue):
+        print("Thread starting ", req.ue_id)
+        attachTime = random.uniform(self.attach_delay_t0, self.attach_delay_t1)
+        attachTimer = threading.Timer(attachTime, self.handle_attach_timer, args=(req,))
+        attachTimer.start()
+
+        while True:
+            message = queue.get()
+            self.ue_handle_msg(message)
+
+        print("Thread finishing ", req.ue_id)
+
+    @staticmethod
+    def get_ue_id(msg):
+        if msg.msg_type == s1ap_types.tfwCmd.UE_AUTH_REQ_IND.value:
+            m = msg.cast(s1ap_types.ueAuthReqInd_t)
+            return m.ue_Id
+        elif msg.msg_type == s1ap_types.tfwCmd.UE_SEC_MOD_CMD_IND.value:
+            m = msg.cast(s1ap_types.ueSecModeCmdInd_t)
+            return m.ue_Id
+        elif msg.msg_type == s1ap_types.tfwCmd.UE_ESM_INFORMATION_REQ.value:
+            m = msg.cast(s1ap_types.ueEsmInformationReq_t)
+            return m.ue_Id
+        elif msg.msg_type == s1ap_types.tfwCmd.INT_CTX_SETUP_IND.value:
+            m = msg.cast(s1ap_types.FwNbIntCtxSetupInd_t)
+            return m.ueId
+        elif msg.msg_type == s1ap_types.tfwCmd.UE_EMM_INFORMATION.value:
+            m = msg.cast(s1ap_types.ueEmmInformation_t)
+            return m.ue_Id
+        elif msg.msg_type == s1ap_types.tfwCmd.UE_ATTACH_ACCEPT_IND.value:
+            m = msg.cast(s1ap_types.ueAttachAccept_t)
+            return m.ue_Id
+        elif msg.msg_type == s1ap_types.tfwCmd.UE_CTX_REL_IND.value:
+            m = msg.cast(s1ap_types.FwNbUeCtxRelInd_t)
+            return m.ueId
+        elif msg.msg_type == s1ap_types.tfwCmd.UE_IDENTITY_REQ_IND.value:
+            m = msg.cast(s1ap_types.ueIdentityReqInd_t)
+            return m.ue_Id
+
+        return 0
+
+    def test_continuous_random_attach(self):
+        num_ues = 100
+
+        # These specify the attach rate as well as the duration of each attach
+        # Actual value is uniformly distributed between t0 and t1
+        self.attach_delay_t0 = 1
+        self.attach_delay_t1 = 10
+        self.attach_duration_t0 = 10
+        self.attach_duration_t1 = 20
+
+        threads = []
+        queues = []
+        self._s1ap_wrapper.configUEDevice(num_ues)
+
+        for index in range(num_ues):
+            req = self._s1ap_wrapper.ue_req
+            print("Create and start thread ", index)
+            q = queue.Queue(maxsize=10)
+            x = threading.Thread(target=self.ue_thread_function, args=(req, q))
+            threads.append(x)
+            queues.append(q)
+            x.start()
+
+        while True:
+            response = self._s1ap_wrapper.s1_util.get_response()
+            ueId = self.get_ue_id(response)
+            if ueId > 0:
+                queues[ueId-1].put(response)
```

Comment:
are we using this queue for any action? it looks like we are only storing up to 10 responses per UE or am I misreading it?

Reply:
Each thread is a consumer of its corresponding queue. I dont see this queue filling up with more than 10 messages per UE